src/backend/distributed/commands/dependencies.c

```diff
@@ -0,0 +1,164 @@
+/*-------------------------------------------------------------------------
+ *
+ * dependencies.c
+ *    Commands to create dependencies of an object on all workers.
+ *
+ * Copyright (c) 2019, Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+
+#include "catalog/dependency.h"
+#include "catalog/objectaddress.h"
+
+#include "distributed/dist_catalog/dependency.h"
+#include "distributed/connection_management.h"
+#include "distributed/worker_manager.h"
+#include "distributed/metadata_sync.h"
+#include "distributed/commands.h"
+#include "distributed/remote_commands.h"
+#include "distributed/dist_catalog/distobject.h"
+
+static List * GetDependencyCreateDDLCommands(const ObjectAddress *dependency);
+
+
+/*
+ * EnsureDependenciesExists finds all the dependencies that we can distribute and makes
+ * sure these are available on all workers. If not available they will be created on the
+ * workers via a separate session that will be committed directly so that the objects are
+ * visible to potentially multiple sessions creating the shards.
```

Comment:
add note that entries in pg_dist_object are created in this transaction, hence objects could already exist on the worker before